processor/src/test/java/org/mapstruct/ap/test/bugs/_2673/Issue2673Mapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2673;
+
+import java.util.Optional;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Ben Zegveld
+ */
+@Mapper
+public interface Issue2673Mapper {
```

Comment:
Let's go with some more descriptive names than `a`, `b`.

Reply:
renamed them to `primitive`, `nonPrimitive` respectively.